pil-analyzer/tests/side_effects.rs

```diff
@@ -100,3 +100,31 @@ fn fixed_with_constr_type() {
     let input = "let x: col = constr |i| 2;";
     analyze_string::<GoldilocksField>(input);
 }
+
+#[test]
+fn set_hint() {
+    let input = r#"
+    namespace std::prover;
+        let set_hint = 8;
+        enum Query { Hint(fe), None, }
+    namespace N(16);
+        let x;
+        std::prover::set_hint(x, query |i| std::prover::Query::Hint(1));
```

Comment:
is `set_hint` just a different way of doing `col ... query` or does it do/allow different things?

Reply:
the `col ... query` syntax is only valid in pil at (outer) statement level. In the long run, we want to get rid of that and only have the syntax that is also available inside functions. So essentially, `set_hint` allows you to set the query function for a witness column from inside a function.